test/integration/kubernetes_deploy_test.rb

```diff
@@ -384,8 +422,27 @@ def test_output_when_switching_labels_incorrectly
       web["spec"]["template"]["metadata"]["labels"] = { "name" => "foobar" }
     end
     assert_equal false, success
-    assert_logs_match("one of your templates is invalid")
-    assert_logs_match(/The Deployment "web" is invalid.*`selector` does not match template `labels`/)
+
+    assert_logs_match_all([
+      "one of your templates is invalid",
+      /The Deployment "web" is invalid.*`selector` does not match template `labels`/
+    ])
+  end
+
+  def test_can_deploy_deployment_with_zero_replicas
+    success = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      web = fixtures["web.yml.erb"]["Deployment"].first
+      web["spec"]["replicas"] = 0
```

Comment:
A use case of zero replicas that I remember is:

1. Create a deployment with unspecified `replicas` field
2. The replicas number is scaled dynamically (can sometimes be zero)
3. At the moment of the deploy it's zero

I see the test works with YAML that explicitly sets `replicas` to zero, but would it also support the case of dynamic `replicas` value set to zero?